### DISCUSSION TITLE
[Suggestion] Minimized version of the Online Players window

### DISCUSSION THREAD

**Discussion by TrueBrain** [OpenTTD/OpenTTD#9504] [Features]
The current Online Players window takes up space for things like "server info" and "player info". This information is mostly static.

Some players have indicated they would appreciate if this information could be hidden in some way, as they are not interested in that information after an initial look. During design of the Online Players window this also came up, and we decided at that point to postpone this feature till we have a bit more information how people use it / what they would like.

The tricky part about this request is, that 10 people have 11 opinions how to go about this. And it has the tendency to get out of hand real quick, with some strong-voiced opinions being thrown in the mix. We have to be careful to not design this for the 0.001%, but come up with something most players enjoy / can use.

Additionally, the GUI framework of OpenTTD is limited in the amount of visual elements it supports, other than manually drawing them (which we should with care, as maintaining those is a pita).

My suggestion is to come up with something simple, both in implementation (so reusing existing GUI elements) and in design, and make a proposal for that.
This discussion is on one hand to collect information what people would appreciate, and on the other hand to allow for design proposal to be made for us to choose from.
Once settled on what would be appropriate, someone can make a Pull Request for it to implement it in OpenTTD itself.

<sub>The default "just to be clear" note: despite me making this discussion doesn't mean I am interested in building this GUI ;) I merely relay information I received from various members of the community to a single place and suggest a way of working :)</sub>

**Comment by PikkaBird** [comment]
To clarify, as I'm "some players", previously I liked to leave the window open and pinned while running multiplayer games, to keep track of who was on the server. See screenshot below. This isn't really convenient with the new window that takes up much more space with the server information etc.

![Winner Transport, 1979-03-29](https://user-images.githubusercontent.com/39507755/130349705-67cd3f6c-9ff1-4604-b76f-57e350774b43.png)

**Comment by LC-Zorg** [comment]
This oversized online players window problem is variable and depends on the resolution of your monitor and the GUI size used. In some cases, the current size of the window will not be a big or even a problem. A lot also depends on where are you playing. If it's a not very busy server with a maximum of a few companies and players, the problem won't be too big. However, in the case of playing on more popular servers, where there are almost always a dozen companies and sometimes even 20-30 players, then this size of the window becomes a significant problem if someone like PikkaBird or myself likes to see who is currently in the game . Personally, I cannot imagine playing online without this window, and as I play almost exclusively online, that's why my reaction to this irreversible change was not very friendly. ;)

A while back I tried to do some research on how popular this usage is among other players and it looks like it's not as popular as I thought. This seems to be mainly because many players were not aware of the existence of this window at all. In general, however, it was that the more experienced the players, the more common it was to use this window. With version 12.0 the popularity of this solution may increase due to the fact that this window will be opened automatically after entering the server (thanks on my part). Time will tell what the reactions of other players will be. I think that among more experienced players they may not be very favorable.

In any case, I would have a few comments and ideas about what can be improved or done. :)

(Not everything is directly related to the window size, some are general improvements, but may be important for consistency of solutions if a reduced version of the window will be added.)

### Improvements for the current window:
**1. Allow a smaller minimum window width** - it will be especially useful for GUI x2 and standard font size (such settings are used by PikkaBird, for example, but I've also seen them in many screenshots, so it's probably quite popular). Especially with this setting the window is unnecessarily wide, but also with the standard GUI size it could be smaller. The visibility of full nicknames and company names or a small window size would be worth leaving to the player's choice. If he wants to see only two letters from company names and nicknames - let him do it.
![ZZ narrower window +](https://user-images.githubusercontent.com/67662046/135156872-3feebd66-c605-432a-88c2-db20db23de20.png)

**2. The height of each bar 2 pixels lower (default GUI size)** - I have doubts, because it would not improve readability, but it would certainly have a significant impact on the size of the window. The reduced stripes would be the same height as in other interface elements (e.g. online content window) - they are now bigger. (Perhaps it is possible to add a general setting to the game that would increase the line spacing everywhere?)
**3. Reducing parts of information about the server and changing the nickname.**
**A. Add a "Shade window" button** - it would hide parts with server and player information. This behavior is slightly different from the other windows. The downside of this solution is the inability to collapse the entire window, but it doesn't seem that important either, because it is just as easy to open it again in the usual way.
**B. Add an "I" button** - it would have the same function as described above, but you would still be able to use the button above to collapse the entire window.
**C1. Reducing the server part to one line** - I presented this solution earlier in my sketches. In my opinion, it has quite an interesting advantage, because you can use icons on it that add quite important functions such as:
- general information about the server important from the point of view of the gameplay, completed automatically and /or entered manually by the admin (icon with the letter I)
- server rules - there is often no way to explain to someone that he is doing something against the general rules, in which case other players could recommend someone to press this icon, others will press themselves out of curiosity.
- speak to all - especially at the beginning of the game, it may be a problem for some players, the more that the description of the functions of the keys is not easily accessible, later this icon doesn't bother you, and for some it may even be convenient, in my opinion it is also an element that improves interface consistency - you can speak to single player, you can speak to company and... you can speak to all.
- server settings - visible and available only to the admin; would open a separate window where all server settings could be changed.
You can use shading to distinguish the server bar from the others without increasing the window size.

**C2. Moving the nickname change function to the icon in the player bar** - I don't think that you need to devote so much space to this function. The pencil icon function should probably be known to every or almost every online player. If it will be on the bar with the player's nickname, you will probably not need to explain what it is for.
What is important in these solutions is the different color of the icons on the server and player bar (as well as the player's company). They are such that, on the one hand, the player can see his position more easily, and on the other hand, so that he can see the most important functions more easily.
![ZZ reducing server and player parts - admin view](https://user-images.githubusercontent.com/67662046/135159614-8d66cf21-8970-4bf9-b33f-09cfd039ce4b.png)

**4. Colorful names for the player and admin** - I have two arguments here; the first is a matter of interpreting the assigned icons, which for some players may be offensive, because it looks like the admin is the king here, and the player is the subject. Replacing the icons with colors on the one hand would neutralize these interpretations, and on the other, improve visibility. Another advantage of this solution is the possibility of assigning other colors to, for example, AI players, friends or players with a registered nickname. In addition, the company founded by or in which the player is would also be marked with the same color. The white marking (New company) would also be consistent here, because it also directly applies to the player.
**5. "(New company)" bar with a blank icon with no color** - gray is the company's color, so using the same color is misleading.
![ZZ new company](https://user-images.githubusercontent.com/67662046/135160870-bddbad4c-63a6-4dd5-b9a9-6027df032ea5.png)

**6. "Spectators" bar without the company's color icon and on a shaded background** - now it looks like one of the companies, but it shouldn't. After all, anyone can create a gray company called Spectators. It's even fun, but also pointless and can be confusing and even appear to be a bug. Also the move to spectators button could be gray.
![ZZ spectators](https://user-images.githubusercontent.com/67662046/135160905-86eea9ab-c303-4278-8699-c484840f7b9c.png)

**7. Self-widening / shrinking window** - similar to the previous window, this could also spontaneously expand and decrease as the number of companies and players present in the game changes. Otherwise, when the list is full, it is possible for a long time not to notice someone's presence, and it may be, for example, someone known, whom you would like to invite to the company. Here you could add a setting for the minimum and especially the maximum number of lines so that the window size doesn't exceed the player's preferences. It can also happen that pressing the size memory button will take this into account as the maximum size and the window will shrink anyway hiding blank lines.
**8. Ordering companies by numbering** - placing the player's company always in the first position introduces a bit of confusion on the list because when the player changes companies they change their order. It's a bit like the company joining the player, not the player joining the company. Each player also sees this list differently.
**9. "New company" button at the bottom of the window** - it is related to the above point and would be justified in case you wanted the order of companies on the list to always keep the numbering. Then the location of this button at the top of the window would not be the best.
![ZZ alternative new company button](https://user-images.githubusercontent.com/67662046/135161942-b3d0cf85-dbed-48fd-af4c-3c6480ec291a.png)

### New smaller window

Despite the above suggestions, even if all about the size reduction were taken into account, the window in many cases will still be unnecessarily large and obscure the view of the game.
Here I apologize in advance. I don't know what is difficult or time-consuming to do and what is easy and quick. Anyway, it was not a problem for me to do this in Paint. ;)

This window wouldn't have to have all the features from the large version. Its main function would be to inform the player about who is online. Besides, the window would only contain these basic communication functions.

![ZZ versions of online players window 1 0](https://user-images.githubusercontent.com/67662046/135164558-f815f779-80af-4e83-910e-0966419a7f2f.png)

### Version 1
It is in fact an improved version of the existing window with added elements that improve consistency with the large window.

**0. The window doesn't display company names** - it is the biggest space saving here. From my point of view, information about the companies on the server is not particularly important, because it is something that is often even unchanged throughout the game.
**1. Switch between large and small windows** - something that players already know from the financial window and I think such a solution would be very intuitive.
**2. Server name at the top of the window, spectators at the bottom** - the "server" and "client" titles, which would be unnecessary in this arrangement, would be removed from the current window. The layouts of large and small windows would be consistent.
**3. Color marking of the player and admin** - it is simple, very legible and doesn't take up space, but icons could also be.
**4. "(New company)" button**
A. At the top of the list - keep the same layout as in the large window
B. At the bottom of the list - a different layout I would prefer for a large window as well.
**5. "Spectators" bar** - pressing it takes you to viewers
**6. Private message** - just press the nickname of the player and write, no other small window will appear anymore
**7. Speak to company** - just press the icon with the company's color
**8. Speak to spectators** - this function would not be in this window or pressing an empty field where the company color icon would be would make it possible.
**9. Change of nickname** - there would be no such possibility here. This is not a frequently used feature, and should the player wish to do so, they can easily switch the window view.

**Version 2**
The same functionality as version 1, but here with a transparent background, the same as under the chat part.

Finally, one more graphic to justify (or refute) the assumptions presented earlier. :)
![Multiplayer window v1 4](https://user-images.githubusercontent.com/67662046/135165637-56947a82-ee59-4907-9d4a-40b4a672c077.png)

**Reply by nielsmh** [reply]
Just as a note, the "Shade window" button that simplifies the window without reducing it to caption-only was something I experimented with for the Frame Rate window. The Finances window does work this way. I think one of the things I concluded was that "shading" definitely needs to make the window smaller in both dimensions, and the shrunk window should be wide rather than tall.
